Implemented `linalg.fused` wrapper operation

KFAFSP opened the pull request.

oowekyala commented on mlir/include/mlir/Dialect/Linalg/Passes.td:
```diff
@@ -210,13 +210,27 @@ def LinalgUnfuse : Pass<"linalg-unfuse", "func::FuncOp"> {
+  ];
+  let options = [
+    ListOption<"strategyString", "strategy", "std::string",
+               "Declarative fusion strategy string.", "llvm::cl::ZeroOrMore">
```
```suggestion
               "Declarative fusion strategy string."
               "The format accepts a comma-separated list of tactics among "
               "seed,producers,producers*,consumers,consumers*,dissolve "
               "plus a filter in parens, which can take set of any of the categories "
               "const,generic,el,act,conv,pool,pad,bcast. Eg seed(conv)", "llvm::cl::ZeroOrMore">
```

oowekyala replied:
According to the [tablegen reference](https://releases.llvm.org/9.0.0/docs/TableGen/LangRef.html#grammar-token-simplevalue), this is the correct syntax:

> Multiple adjacent string literals are concatenated like in C/C++. The value is the concatenation of the strings.

KFAFSP commented on mlir/include/mlir/Dialect/Linalg/IR/Linalg.h:
```diff
@@ -32,6 +32,8 @@
 namespace mlir {
 namespace linalg {
 
```
```suggestion

/// Returns a singelton FallbackModel for the OperatorClassInterface.
///
/// This FallbackModel is attached to all linalg dialect operations so that we
/// don't have to change their definitions.
```

KFAFSP commented:
@ljfitz There is no way to pass defaults to a ListOption, but I had added a TODO to the pass implementation to do just that. I added the current "state of the art" default to that.
